Add `JSGenerateToNativeObject` attributes to `Text`

tetsuharuohzeki opened the pull request:
#### 062294f6788fc6ce5688e47ffe1c23955cacbff1
<pre>
Add `JSGenerateToNativeObject` attributes to `Text`
<a href="https://bugs.webkit.org/show_bug.cgi?id=243655">https://bugs.webkit.org/show_bug.cgi?id=243655</a>

Reviewed by Ryosuke Niwa.

`HTMLSlotElement.assign()` takes `undefined assign((Element or Text)...nodes)`
as its signature.
This requires to call `convertVariadicArguments()` that calls
`JSToWrappedOverloader::toWrapped()` that requires
`WrapperType::toWrapped()`

To fullfill this requirement, we need generate `JSText::toWrapped()` to `Text`
by adding `JSGenerateToNativeObject`.

* Source/WebCore/dom/Text.idl:

Canonical link: <a href="https://commits.webkit.org/253202@main">https://commits.webkit.org/253202@main</a>
</pre>

rniwa commented on Source/WebCore/dom/Text.idl:
```diff
@@ -19,6 +19,7 @@
 
```
No copyright change?

tetsuharuohzeki replied:
I'll change it.

tetsuharuohzeki replied:
addressed in https://github.com/WebKit/WebKit/pull/3099/commits/9de2b2c865e3b75444db4d1af2e7cf8226c10320

rniwa reviewed (approved):
r=me

tetsuharuohzeki committed:
Add `JSGenerateToNativeObject` attributes to `Text`
https://bugs.webkit.org/show_bug.cgi?id=243655

Reviewed by Ryosuke Niwa.

`HTMLSlotElement.assign()` takes `undefined assign((Element or Text)...nodes)`
as its signature.
This requires to call `convertVariadicArguments()` that calls
`JSToWrappedOverloader::toWrapped()` that requires
`WrapperType::toWrapped()`

To fullfill this requirement, we need generate `JSText::toWrapped()` to `Text`
by adding `JSGenerateToNativeObject`.

* Source/WebCore/dom/Text.idl:

Canonical link: https://commits.webkit.org/253202@main
